### ISSUE TITLE
Improve mechanizm of searching plugins and resources in Linux and macOS

### ISSUE THREAD

**Issue by tehnick** [psi-im/psi#589]
Resources:
* Translation files
* Myspell dictionaries
* Icons
* etc.

For macOS this is necessary for launching of `Psi+.app` and `Psi.app` which are not installed to `/Applications/`.

For GNU/Linux this is necessary for correct work of program packaged in AppImage, snap or flatpak, where the mount points may be changed by different reasons or even may be unpredictable by design.

For GNU/Linux there are two possible solutions:
* Use environment variables for each type of resources and one common script for launching program.
* Use relative paths which are based on path to executable file. For example:
```shell
$ tree -d ~/.local/bin/Psi+/
/home/user/.local/bin/Psi+/
├── bin
├── lib
│   └── psi-plus
│       └── plugins
└── share
    ├── applications
    ├── icons
    │   └── hicolor
    │       ├── 128x128
    │       │   └── apps
    │       ├── 16x16
    │       │   └── apps
    │       ├── 32x32
    │       │   └── apps
    │       ├── 48x48
    │       │   └── apps
    │       └── 64x64
    │           └── apps
    ├── pixmaps
    └── psi-plus
        ├── certs
        ├── iconsets
        │   └── roster
        ├── myspell
        │   └── dicts
        ├── sound
        └── translations
```

I prefer the second variant.

**Comment by tehnick** [comment]
Fixed for GNU/Linux: 93709690634db4190128b57cd51fd600736bbe83

**Comment by tehnick** [comment]
I have tested launching of program from different non-standard paths in macOS and everything worked fine.
